### PR TITLE
Add `web3-profile` extension

### PR DESCRIPTION
**Pull request by peduarte**
## Description

An extension to search for ENS names and display related information. Such as avatar, description, social URLs, etc.

The main command is called `search-web3-profile`.

## Screencast

https://user-images.githubusercontent.com/372831/188460029-bc9e2eac-018c-48f9-b562-1100834433ac.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/web3-profile/src/search-web3-profile.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ export default function Command() {
   return (
     <List isLoading={isLoading} searchBarPlaceholder="ENS search" onSearchTextChange={setSearchTerm} throttle>
       <List.EmptyView
+        title={searchTerm.length ? "No Results" : "Search For ENS Name"}
```
Comment (peduarte) [inline extensions/web3-profile/src/search-web3-profile.tsx:28]: ahh yeah, thanks for that. totally forgot 😄

Comment (pernielsentikaer) [reply extensions/web3-profile/src/search-web3-profile.tsx]: Could we do something about the logo here? Can you make a logo in `64x64px` instead? I don't think we need the text here 🙂

File: extensions/web3-profile/src/search-web3-profile.tsx [hunk 2]
```diff
@@ -45,6 +45,7 @@ export default function Command() {
             <ActionPanel>
               <Action.Push
                 title="Show Profile"
+                icon={Icon.AppWindowSidebarLeft}
```
Comment (peduarte) [inline extensions/web3-profile/src/search-web3-profile.tsx:48]: nice touch 👌  think we may need to import `Icon` here

File: extensions/web3-profile/package.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  "keywords": [
+    "web3",
+    "ens",
+    "ethereum"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "search-web3-profile",
+      "title": "Search Web3 Profiles",
```
Comment (pernielsentikaer) [inline extensions/web3-profile/package.json:20]:
```suggestion
      "title": "Search Profiles",
```
Maybe move Web3 to subtitle?

File: extensions/web3-profile/package.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      "subtitle": "View address, ENS records, and more.",
```
Comment (pernielsentikaer) [inline extensions/web3-profile/package.json:21]:
```suggestion
      "subtitle": "Web3",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @peduarte 👋 

Thanks for an awesome extension! I added a few small things to make the UI/UX a bit better and have a few suggestions 🙂

- Could you make sure that all metadata images are taken with the screenshot tool so they look the same

- Is it possible to cache results for a given amount of time or allow the user to pass an API key?

I saw this message in the console
```16:03:54.926 ========= NOTICE =========
16:03:54.926 Request-Rate Exceeded  (this message will not be repeated)
16:03:54.926 
16:03:54.926 The default API keys for each service are provided as a highly-throttled,
16:03:54.926 community resource for low-traffic projects and early prototyping.
16:03:54.926 
16:03:54.926 While your application will continue to function, we highly recommended
16:03:54.933 signing up for your own API keys to improve performance, increase your
16:03:54.933 request rate/limit and enable other perks, such as metrics and advanced APIs.
16:03:54.934 
16:03:54.934 For more details: https://docs.ethers.io/api-keys/
16:03:54.934 ==========================
```

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/web3-profile/src/search-web3-profile.tsx (hunk 2 above)
Comment (pernielsentikaer) [reply extensions/web3-profile/src/search-web3-profile.tsx:48]: Yes, good catch - I thought that I added it 😃

File: extensions/web3-profile/package.json (hunk 4 above)
Comment (peduarte) [reply extensions/web3-profile/package.json:21]: sounds good!

**Comment by pernielsentikaer** [comment]
I have to go away for a few hours; I will check a bit later again 👍 🤘

**Comment by peduarte** [comment]
@pernielsentikaer thanks for taking the tie for an excellent review! 🙏 

> Could you make sure that all metadata images are taken with the screenshot tool so they look the same
Done! I read that in the docs but didnt understand it was mandatory, thought it was just an option. But I agree it looks better!

> Is it possible to cache results for a given amount of time or allow the user to pass an API key?
So this was my plan for V2 😬 I started setting things in Local Storage, but would need to re-think the structure of things, so decided to ship now and see how it goes, then continue to improve. The public API key will work 99% of time, maybe it'll be slower than usual at times due to throttling but should work

File: extensions/web3-profile/src/search-web3-profile.tsx (hunk 1 above)
Comment (peduarte) [reply extensions/web3-profile/src/search-web3-profile.tsx]: to be honest, i like it how it is! the overflowed text on the logo gives it a bit of character and sets a nice vibe imo

but if this is breaking any strict guidelines, happy to change it 👍

File: extensions/web3-profile/src/search-web3-profile.tsx (hunk 2 above)
Comment (peduarte) [reply extensions/web3-profile/src/search-web3-profile.tsx:48]: running this locally, and i cant see any icon (`Icon.AppWindowSidebarLeft`) in the list 
CleanShot 2022-09-05 at 20 37 08@2x

File: extensions/web3-profile/src/search-web3-profile.tsx [hunk 5]
```diff
@@ -25,7 +25,13 @@ export default function Command() {
...
     <List isLoading={isLoading} searchBarPlaceholder="ENS search" onSearchTextChange={setSearchTerm} throttle>
       <List.EmptyView
-        title={searchTerm.length ? "No Results" : "Search For ENS Name"}
+        title={
+          isLoading
+            ? "Searching..."
+            : searchTerm.length < 4 && ensSuggestions.length > 0
+            ? "Search For ENS Name"
+            : "No Results"
+        }
```
Comment (peduarte) [inline extensions/web3-profile/src/search-web3-profile.tsx:34]: just testing this change, but I'm unable to ever see the `"Search For ENS Name"` case in the UI

was your intent to display that text when the search bar was empty?

File: extensions/web3-profile/src/search-web3-profile.tsx (hunk 2 above)
Comment (pernielsentikaer) [reply extensions/web3-profile/src/search-web3-profile.tsx:53]: That's strange, it should be a valid icon for quite some time now, does `npm install` do something?

File: extensions/web3-profile/src/search-web3-profile.tsx (hunk 5 above)
Comment (pernielsentikaer) [reply extensions/web3-profile/src/search-web3-profile.tsx:39]: Ah. could you try again? Also updated the logo, there was a lot of space due to the size - feel free to modify :)

Comment (peduarte) [reply extensions/web3-profile/src/search-web3-profile.tsx:40]: The empty string on line 38 is causing the UI (the logo) to shift between states, video below:

https://user-images.githubusercontent.com/372831/188591908-9a480391-3d7d-4100-aa8e-fb2b2df50fe8.mp4

I've made some changes to remove the `description` and update the `title` with necessary instructions. I also refactored the `title` condition to make it easier to reason about. demo below:

https://user-images.githubusercontent.com/372831/188594329-7096bb8b-44bd-427f-8601-f70b4bf2732d.mp4

Comment (pernielsentikaer) [reply extensions/web3-profile/src/search-web3-profile.tsx]: I like it, awesome @peduarte 🔥

**Comment by pernielsentikaer** [comment]
@peduarte do you want to make a dark version of [web3-profile-icon.png](https://github.com/raycast/extensions/pull/2803/files#diff-2114bb191fbdc61c0825c15f1a77c3ef684b8302734d98031622123600c3b7a9) too?

**Comment by peduarte** [comment]
@pernielsentikaer i wasnt planning on doing that because i think it works on both modes:

CleanShot 2022-09-06 at 12 07 57@2x

image


what do you think?

**Comment by pernielsentikaer** [comment]
Let's keep it, I think we're about ready 👏

**Comment by peduarte** [comment]
LFGGGGG

**Pull request merged**